### PR TITLE
Fix deadlock when responding to contacts sync messages.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Messages/OWSMessageManager.m [hunk 1]
```diff
@@ -606,21 +606,30 @@ - (void)handleIncomingEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
         }
     } else if (syncMessage.hasRequest) {
         if (syncMessage.request.type == OWSSignalServiceProtosSyncMessageRequestTypeContacts) {
-            OWSSyncContactsMessage *syncContactsMessage =
```
Comment (charlesmchen) [inline SignalServiceKit/src/Messages/OWSMessageManager.m]: Mostly a whitespace change.

Comment (michaelkirk) [reply SignalServiceKit/src/Messages/OWSMessageManager.m]: Is there a reason we don't want to do the same treatment for group sync requests?

File: SignalServiceKit/src/Security/OWSRecipientIdentity.m [hunk 2]
```diff
@@ -123,6 +123,35 @@ + (YapDatabaseConnection *)dbConnection
     return sharedDBConnection;
 }
 
+- (void)saveWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
```
Comment (michaelkirk) [inline SignalServiceKit/src/Security/OWSRecipientIdentity.m]: Nice. 👌

File: SignalServiceKit/src/Messages/OWSMessageManager.m (hunk 1 above)
Comment (charlesmchen) [reply SignalServiceKit/src/Messages/OWSMessageManager.m]: `[OWSSyncGroupsMessage buildPlainTextAttachmentDataWithTransaction:]` already takes a transaction and none of its inner workings (e.g. `[OWSGroupsOutputStream writeGroup:]` seem unsafe.

**Pull request merged**